src/test/java/org/sagebionetworks/bridge/sdk/integration/ExternalIdsTest.java

```
@@ -0,0 +1,86 @@
+        for (int i=0; i < LIST_SIZE; i++) {
+            identifiers.add(prefix+RandomStringUtils.randomAlphabetic(10));
+        }
+        DeveloperClient client = developer.getSession().getDeveloperClient();
+        try {
+            client.addExternalIds(identifiers);
+            
+            PagedResourceList<ExternalIdentifier> page1 = client.getExternalIds(null, PAGE_SIZE, prefix, null);
+            assertEquals(PAGE_SIZE, page1.getItems().size());
+            assertEquals(LIST_SIZE, page1.getTotal());
+            assertNotNull(page1.getOffsetKey());
+            
+            String offsetKey = page1.getOffsetKey();    
+            PagedResourceList<ExternalIdentifier> page2 = client.getExternalIds(offsetKey, PAGE_SIZE, prefix, null);
+            assertEquals(PAGE_SIZE, page2.getItems().size());
+            assertEquals(LIST_SIZE, page2.getTotal());
+            assertNull(page2.getOffsetKey()); // no more pages
+            
+            // pageKey test. two pages should have no members in common;
+            assertTrue(Collections.disjoint(page1.getItems(), page2.getItems()));
+
+            // assignment filter test
+            page1 = client.getExternalIds(null, null, prefix, Boolean.FALSE);
+            assertEquals(LIST_SIZE, page1.getTotal());
+            
+            page1 = client.getExternalIds(null, null, prefix, Boolean.TRUE);
+            assertEquals(0, page1.getTotal());
+            
+            // different idFilter test (than the use of idFilter in all the tests. This time nothing should match.
+            page1 = client.getExternalIds(null, PAGE_SIZE, RandomStringUtils.randomAlphabetic(5), null);
+            assertEquals(0, page1.getTotal());
+        } finally {
+            client.deleteExternalIds(identifiers);
+
+            PagedResourceList<ExternalIdentifier> page = client.getExternalIds(null, null, prefix, null);
+            assertEquals(0, page.getTotal());
```

Similarly, we should move the check that getExternalIds() returns nothing to after the finally block. This way, if the test fails, we're unlikely to have a secondary failure in the finally block that masks the original failure.